packages/graphics/mesa/package.mk

```diff
@@ -90,6 +98,20 @@ pre_configure_target() {
   if [ "$DISPLAYSERVER" = "x11" ]; then
     export LIBS="-lxcb-dri3 -lxcb-dri2 -lxcb-xfixes -lxcb-present -lxcb-sync -lxshmfence -lz"
   fi
+
+  # Temporary hack (until panfrost evolves) to use 64-bit pointers in structs passed to GPU
+  # even if userspace is 32-bit. This is required for Mali-T820 to work with mesa built for
+  # arm userspace. The hack does not affect building for aarch64.
+  if [ "$PROJECT" = "Amlogic" ]; then
```

Comment:
Same as above.

Reply:
The hack only applies to T8xx GPU so it should probably be reworked to check `MALI_FAMILY` for T8xx.

Reply:
i'd rather not include hacks like this in the package. If you need it then please use a patch.

Reply:
We used a patch previously but it had to be amended every time something in Panfrost changed, this is a no-maintenance approach.

Reply:
> We used a patch previously but it had to be amended every time something in Panfrost changed, this is a no-maintenance approach.

Yes, that's how patches work 😸 It will need to be updated when things change.

Reply:
This needs to remain until the issue is properly resolved upstream. Until then we already have an indecent number of things to track/update frequently and having to constantly rebase some mesa patches instead of a simple githash change is just forcing "makework" activity on a small group of people who are already stretched for time. Yes it's a hack. Yes it will be gone as soon as we manage to get the Panfrost devs to remember 32-bit userspace is a real thing.